application/src/test/java/org/opentripplanner/transit/service/DefaultTransitServiceTest.java

```diff
@@ -241,6 +241,24 @@ void getRealtimeTripTimes() {
     );
   }
 
+  @Test
+  void getTripTimesOnNoServiceDayWithUseScheduleOption() {
+    Instant midnight = ServiceDateUtils.asStartOfService(
+      SERVICE_DATE,
+      service.getTimeZone()
+    ).toInstant();
+
+    assertEquals(
+      Optional.of(
+        List.of(
+          new TripTimeOnDate(REALTIME_TRIP_TIMES, 0, REAL_TIME_PATTERN, SERVICE_DATE, midnight),
+          new TripTimeOnDate(REALTIME_TRIP_TIMES, 1, REAL_TIME_PATTERN, SERVICE_DATE, midnight)
+        )
+      ),
+      service.getTripTimeOnDates(TRIP, SERVICE_DATE, true)
```

Comment:
I think you are testing for a wrong date. `SERVICE_DATE` is a date the service does run, and `NO_SERVICE_DATE` is a date the service does not run.